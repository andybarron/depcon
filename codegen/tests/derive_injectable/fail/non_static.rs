use depcon::*;
use std::marker::PhantomData;

#[derive(Injectable)]
struct NonStatic<'a>(PhantomData<&'a ()>);

fn main() {}
