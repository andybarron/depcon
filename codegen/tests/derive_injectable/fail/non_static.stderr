error[E0478]: lifetime bound not satisfied
 --> tests/derive_injectable/fail/non_static.rs:4:10
  |
4 | #[derive(Injectable)]
  |          ^^^^^^^^^^
  |
note: lifetime parameter instantiated with the lifetime `'a` as defined here
 --> tests/derive_injectable/fail/non_static.rs:5:18
  |
5 | struct NonStatic<'a>(PhantomData<&'a ()>);
  |                  ^^
  = note: but lifetime parameter must outlive the static lifetime
  = note: this error originates in the derive macro `Injectable` (in Nightly builds, run with -Z macro-backtrace for more info)
